src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -324,5 +517,99 @@ internal static byte[] ConstructSequence(params byte[][][] items)
 
             return encodedSequence;
         }
+
+        private static BigInteger ParseOidRid(string oidValue, int startIndex, int endIndex)
+        {
+            if (startIndex > oidValue.Length)
+                throw new CryptographicException();
+
+            if (endIndex == -1)
+            {
+                endIndex = oidValue.Length;
+            }
+
+            Debug.Assert(endIndex > startIndex);
+
+            // The following code is equivalent to
+            // BigInteger.TryParse(temp, NumberStyles.None, CultureInfo.InvariantCulture, out value)
+            // but doesn't require creating the intermediate substring (TryParse doesn't take start+end/length values)
+
+            int position = startIndex;
+            int seed = AtoI(oidValue[position]);
+            BigInteger value = new BigInteger(seed);
+
+            for (position++; position < endIndex; position++)
```

Comment:
Doing a postincrement in the declaration is pretty confusing. Could you just increment it above?